### PR TITLE
Riscv timer

### PR DESCRIPTION
**Pull request by Skallwar**
This PR add support for timer (riscv64 only right now)

This is a first step toward scheduling

### CODE REVIEW

**Review by n1tram1 — APPROVED** [review]

File: src/arch/riscv64/interrupts.rs [hunk 1]
```diff
@@ -2,8 +2,122 @@ use crate::arch::riscv64::registers;
...
+            5 => Self::SupervisorTimer,
+            9 => Self::SupervisorExternal,
+            _ => Self::Platform(code),
+        }
+    }
+}
+
+impl From<InterruptType> for u64 {
+    fn from(itype: InterruptType) -> Self {
+        match itype {
```
Comment (CohenArthur) [inline src/arch/riscv64/interrupts.rs:41]: So we can't do `itype as u64` here even tho it's `repr(u64)`?

**Review by CohenArthur — APPROVED** [review]
LGTM!

File: src/arch/riscv64/interrupts.rs (hunk 1 above)
Comment (Skallwar) [reply src/arch/riscv64/interrupts.rs:41]: I should remove the `repr(u64)`, this is legacy of the `enum` whit assign discriminant.

The `itype as u64` was no working with the nightly feature required for discriminant with `enum` containing sum types. There was some ugly unsafe code so I remove it.

Comment (Skallwar) [reply src/arch/riscv64/interrupts.rs:40]: Fixed by [e60eb21](https://github.com/ks0n/goOSe/pull/95/commits/e60eb21f12707f423ea16cbc00cadec80b88fb5b)

**Review by CohenArthur — APPROVED** [review]
LGTM!

**Pull request merged**